db/migrate/20231030102203_add_original_batch_id_to_samples.rb

```diff
@@ -1,5 +1,13 @@
 class AddOriginalBatchIdToSamples < ActiveRecord::Migration
   def change
     add_column :samples, :original_batch_id, :integer
+
+    Sample.preload(:batch).find_each do |sample|
+      batch = sample.batch || Batch.find_by(batch_number: sample.old_batch_number)
```

Comment:
**suggestion**: maybe search the old batch number through the box? so we'd be 100% sure that the batch is really the original batch, just in case.